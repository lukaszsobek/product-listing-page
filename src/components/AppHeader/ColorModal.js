import React, { Component } from "react";
import { connect } from "react-redux";

import { toggleModal } from "../../actions";

const DropDown = props => {
    if(!props.isOpen) {
        return null;
    }

    return(
        <div className="modal--color__data">
            something
        </div>
    );
}

class ColorModal extends Component {

    render() {
        const { isOpen, toggleModal } = this.props;
        return (
            <div className="modal--color">
                <div
                    className="modal--color__title"
                    onClick={toggleModal}
                >
                    Colors
                </div>
                <DropDown isOpen={isOpen}/>
            </div>
        );
    }
};

const mapStateToProps = state => ({
    isOpen: state.modalState.colorModalVisible
});

const mapDispatchToProps = dispatch => ({
    toggleModal: () => dispatch(toggleModal("ColorModal"))
});

export default connect(mapStateToProps, mapDispatchToProps)(ColorModal);