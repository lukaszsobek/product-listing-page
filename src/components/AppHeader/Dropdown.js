import React from 'react'

const Dropdown = ({
  clearFilter, data, dropdownClass, activeFilters, isOpen, toggleFilter
}) => {
  if (!isOpen) {
    return null
  }

  const activeFilterString = activeFilters.join(',,,')

  const list = data.map((dataItem, key) => {
    const isFilterActive = activeFilterString.indexOf(dataItem) > -1

    return (
      <div
        className={isFilterActive ? 'active' : ''}
        key={key}
        onClick={() => toggleFilter(dataItem)}
      >{dataItem}</div>
    )
  })

  return (
    <div className={dropdownClass}>
      { list }
      { activeFilters.length > 0 &&
        <div
          onClick={clearFilter}
        >...clear</div>
      }
    </div>
  )
}

export default Dropdown
