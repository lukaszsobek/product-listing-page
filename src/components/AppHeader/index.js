import React from "react";

const AppHeader = props => {
    return <p>Header</p>
};

export default AppHeader;