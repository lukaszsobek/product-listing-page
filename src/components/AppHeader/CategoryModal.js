import React from 'react'
import { connect } from 'react-redux'

import {
  updateClearFilterDisplay,
  updateToggleFilterDisplay,
  toggleModal
} from '../../actions'
import Dropdown from './Dropdown'

const CategoryModal = props => {
  const {
    activeFilters,
    clearFilter,
    data,
    isOpen,
    toggleFilter,
    toggleModal
  } = props

  return (
    <div className={
      isOpen
        ? 'modal--category modal--category--open'
        : 'modal--category'
    }>
      <div
        className='modal--category__title'
        onClick={toggleModal}
      >Category</div>
      <Dropdown
        isOpen={isOpen}
        data={data}
        activeFilters={activeFilters}
        toggleFilter={toggleFilter}
        clearFilter={clearFilter}
        dropdownClass='modal--category__data'
      />
    </div>
  )
}

const mapStateToProps = ({ root: state }) => ({
  isOpen: state.modalState.categoryModalVisible,
  data: state.dropdownData.categories,
  activeFilters: state.activeFilters.categories
})

const mapDispatchToProps = dispatch => ({
  toggleModal: () => dispatch(toggleModal('CategoryModal')),
  toggleFilter: (value) => dispatch(
    updateToggleFilterDisplay('categories', value)
  ),
  clearFilter: () => dispatch(
    updateClearFilterDisplay('categories')
  )

})

export { CategoryModal }
export default connect(mapStateToProps, mapDispatchToProps)(CategoryModal)
