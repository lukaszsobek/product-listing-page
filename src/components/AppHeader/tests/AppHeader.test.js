import AppHeader from "../";

describe("AppHeader", () => {
    it("renders",() => {
        expect(AppHeader()).toMatchSnapshot();
    });
});