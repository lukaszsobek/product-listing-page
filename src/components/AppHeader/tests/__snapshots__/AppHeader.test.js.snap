// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader renders 1`] = `
<header>
  <nav
    className="container"
  >
    <div
      className="title"
    >
      Sample Products
    </div>
    <Connect(CategoryModal) />
    <Connect(ColorModal) />
  </nav>
</header>
`;
